.github/workflows/deploy.yml

```
@@ -0,0 +1,24 @@
+name: deploy
+on:
+  release:
+    types: [published]
+
+jobs:
+  build:
+    name: Protoc Test Build and Compile
+    uses: ./.github/workflows/_protoc_build.yml
+  publish:
+    name: Build and Publish
+    needs: build
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
```

v3 of checkout is the latest version.

```suggestion
        uses: actions/checkout@v3
```